02.calendar/calendar.rb

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+
+opt = OptionParser.new
+
+opt.on('-y') {|v| v }
+opt.on('-m') {|v| v }
+
+argv = opt.parse(ARGV)
+
+if argv[0].to_i == 0
+  @month = Date.today.month
+else
+  @month = argv[0].to_i
+end   
+
+if argv[1].to_i == 0
+  @year = Date.today.year
+else
+  @year = argv[1].to_i
+end
+
+def check_first_wday(y = @year, m = @month)
+  @first_wday = Date.new(y, m, 1).wday
+end
+check_first_wday
```

Comment:
ここで check_first_wday のメソッドを定義する理由はどうしてでしょうか？

* 定義してすぐ実行している
* メソッドの内容が１行
* 定義したメソッドを１回しか使っていない

などの理由から、わざわざメソッドを定義する必要はなさそうで

```suggestion
@first_wday = Date.new(@year, @month, 1).wday
```

これだけでよさそうに見えました。

Reply:
ごもっともです。メソッドの書き方を覚えたので使ってみたいという気持ちがあり、メソッドにしなくても動くということに気づいていませんでした。end_dateも類似しているので、first_wdayとend_dateの部分を修正いたします。

Reply:
メソッドの定義をやめ、インスタンス変数の＠を削除、ローカル変数に変更しました。

Reply:
👍